### PR TITLE
Initial commit for NTT Ltd. Cyber Threat Sensor integration

### PR DESCRIPTION
**Pull request by jryberg**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
n/a

## Description
Initial commit

## Minimum version of Demisto
- [ ] 4.5.0
- [x] 5.0.0
- [ ] 5.5.0

Tested with 5.0.0 and 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation 

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

File: Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/fetch_incidents.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+            artifacts_meta.append(result_meta)
+        if 'offset' in results_meta:
+            last_run['offset'] = results_meta['offset']
+        else:
+            last_run.pop('offset', None)
+            last_run['timestamp'] = datetime.now().isoformat()
+
+    incidents = list()
+    for artifact_meta in artifacts_meta:
+        demisto.debug('\nRequesting data for event: %s\n\n' % artifact_meta['event_id'])
```
Comment (reutshal) [inline Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/fetch_incidents.py:135]:
```suggestion
        demisto.debug('\nRequesting data for event: {}\n\n'.format(artifact_meta['event_id']))
```

File: Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/fetch_incidents.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        result_artifact = http_request('GET', '/artifacts/alerts/%s' % artifact_meta['event_id'])
+        incidents.append(item_to_incident(result_artifact['alert']))
+
+    demisto.incidents(incidents)
+    demisto.setLastRun(last_run)
+    return_results('ok')
```
Comment (reutshal) [inline Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/fetch_incidents.py:141]:
```suggestion
```
No need to return anything from this command

File: Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/fetch_incidents.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    """Send the request to API and return the JSON response
+    """
+    r = requests.request('GET', url)
+    if r.status_code != requests.codes.ok:
+        return_error('Error in API call to download %s - %s' % (url, r.text))
+    return r
+
+
+def item_to_incident(item):
+    dt_string = item.get('last_updated')
```
Comment (reutshal) [inline Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/fetch_incidents.py:85]:
```suggestion
    dt_string = item.get('last_updated', '')
```

File: Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/README.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+| CTS.ARADetails | unknown | Details of the recommended actions | 
+| CTS.Signature | unknown | Signature name of detection | 
+| CTS.OccuredTime | unknown | Date and time of occurrence or detection | 
+| CTS.DeviceId | unknown | Id of detecting device | 
+| CTS.TenantId | unknown | Id of the tenant | 
+| CTS.EventID | string | Id of the event | 
+
+
+#### Command Example
+```!fetch_incidents ```
```
Comment (reutshal) [inline Packs/NTT_Cyber_Threat_Sensor/Integrations/fetch_incidents/README.md:56]: Please create a new README file after you remove the `fetch-incidents` command - 
It shouldn't be here :)

**Comment by jryberg** [comment]
Hi @reutshal

Thank you for your feedback.

I will start working with this once I'm back from my annual leave 10th of August. 

Best regards Johan Ryberg

**Comment by reutshal** [comment]
Hi @jryberg,
Are you back from your annual leave?
I tried to contact you on slack but with no success.
I see your PR have some additional changes from our master branch,
I think we should schedule a zoon meeting and solve this for you.
Please share your availability. Thanks!

**Comment by jryberg** [comment]
Hi,

I'm still on vacation but a colleague of mine are trying to fix the
outstanding issues that was found.

I will be back 10th of August.

It's okey if you and Jason have a meeting during the week or we can
schedule a meeting once I'm back.

Best regards Johan Ryberg

**Comment by reutshal** [comment]
@jryberg Whatever is more comfortable for you guys.
I suggest meeting for @jflowersntt in order to fix the PR issues,
and when you're back we will schedule a demo.
@jflowersntt Please sure your availability for a zoom session, I am available most of the week.
Thanks!

**Comment by jflowersntt** [comment]
I am free anytime to discuss this, Please schedule a zoom meeting at your earliest convenience

Thanks
Jason

**Comment by reutshal** [comment]
@jflowersntt Please share an email so I can send the invite

**Comment by jflowersntt** [comment]
Sure thing, My email is Jason.flowers@global.ntt<mailto:Jason.flowers@global.ntt>, please invite ulrik.holmen@global.nt<mailto:ulrik.holmen@global.nt> as well.


Thanks

**Comment by mchasepan** [comment]
@trorabaugh FYI

**Comment by trorabaugh** [comment]
Hi @mchasepan @fvigo we had a full conversation on slack already regarding this in the review channel

**Comment by reutshal** [comment]
Hi everyone, this is a conclusion of tasks after the demo we had:
- [ ] Integration playbook - 
  - [ ] Add generic-polling and remove the `sleep` from the code. You can find a generic polling example in `Panorama Query Logs` playbook.
  - [ ] Add PCAP errors handling according to @David-BMS suggestion. You can find an example in `PCAP Analysis` playbook.
- [ ] Change the pack_metadate email to the support email rather than a personal one
- [ ] Add human-readable entry to `fetch-blobs` command
- [ ] Add to the integration README - The playbook documentation and the fetch-blobs human-readable output.
- [ ] Add a README file to the playbook (can be generated by the `generate-docs` sdk command)

You can check the tasks here once they are done so we will be synchronized.
If you have anything to add please do :)

File: Packs/NTT_Cyber_Threat_Sensor/Integrations/NTT_Cyber_Threat_Sensor/NTT_Cyber_Threat_Sensor.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+"""GLOBALS/PARAMS
+"""
+
+APIKEY = demisto.params().get('APIKEY')
+SOARTOKEN = demisto.params().get('SOARTOKEN')
+BASEURL = demisto.params().get('BASEURL', '').strip('/')
+TENANT_ID = demisto.params().get('TENANT_ID')
+DAYS_BACK = demisto.params().get('DAYS_BACK')
+ITEMS_TO_FETCH = demisto.params().get('ITEMS_TO_FETCH')
+USE_SSL = demisto.params().get('insecure')
```
Comment (reutshal) [inline Packs/NTT_Cyber_Threat_Sensor/Integrations/NTT_Cyber_Threat_Sensor/NTT_Cyber_Threat_Sensor.py:26]:
```suggestion
USE_SSL = not demisto.params().get('insecure')
```

**Comment by jryberg** [comment]
> Looks great, please change the commands name to start with the vendor name as discussed in the demo

Done!

File: Packs/NTT_Cyber_Threat_Sensor/Playbooks/Handle_TD_events_README.md [hunk 6]
```diff
@@ -25,9 +26,9 @@ There are no inputs for this playbook.
...
 | --- | --- | --- |
-| CTS.EventID | Event ID | string |
-| CTS.OccuredTime | Event Time | unknown |
+| CTS.EventID | CTS EventID aka alert / sha | string |
+| CTS.OccuredTime | Timestamp when incident was registered | date |
 
 ## Playbook Image
 ---
-![Handle TD events](Insert the link to your image here)
+![Handle TD events](Insert the link to your image here)
```
Comment (reutshal) [inline Packs/NTT_Cyber_Threat_Sensor/Playbooks/Handle_TD_events_README.md:34]: Please insert the playbook image here.
You can find the instructions here: https://xsoar.pan.dev/docs/integrations/integration-docs#images

**Review by reutshal — APPROVED** [review]

File: Packs/NTT_Cyber_Threat_Sensor/pack_metadata.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+    "name": "NTT Cyber Threat Sensor",
+    "description": "NTT Cyber Threat Sensor integration",
+    "support": "partner",
```
Comment (ShahafBenYakir) [inline Packs/NTT_Cyber_Threat_Sensor/pack_metadata.json:4]: @reutshal Did we receive any info that indicates that they are partners?

**Comment by jryberg** [comment]
Hi, Reut received our partner ID via PM in Slack.

Best regards Johan Ryberg

**Comment by jryberg** [comment]
https://www.paloaltonetworks.com/company/press/2020/ntt_and_palo_alto_networks_expand_strategic_partnership

**Comment by mchasepan** [comment]
> https://www.paloaltonetworks.com/company/press/2020/ntt_and_palo_alto_networks_expand_strategic_partnership

That does not mean we have the right legal protections in place for contributions.... I'll confirm with our teams.

**Pull request merged**